packages/example/src/containers/Dashboard/Dashboard.tsx

```diff
@@ -49,6 +50,21 @@ export const Dashboard = () => {
         }
     }, [api, setMessages]);
 
+    useEffect(() => {
+        (async () => {
+            const isConnected = sessionStorage.getItem('metamask-snap');
+            if (isConnected) {
+                const installResult = await installFilecoinSnap();
+                if (installResult.isSnapInstalled) { 
+                    dispatch({
+                        type: MetamaskActions.SET_INSTALLED_STATUS,
+                        payload: {isInstalled: true, snap: installResult.snap}
+                    });
+                }
+            }
+        })();
+    } , [dispatch]);
+
```

Comment:
Seems that [MetaMaskConnector.tsx](https://github.com/ChainSafe/filsnap/pull/140/files#diff-f2ee584f4eba77a15b2c91166b853fea53e228bff16deffb7801d1d77677bb93) is better place for this logic